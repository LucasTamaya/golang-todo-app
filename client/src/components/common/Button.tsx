interface Props {
  title: string;
  bgColor: string;
  onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => any;
}

const Button: React.FC<Props> = ({ title, bgColor, onClick }) => {
  return (
    <button
      className={`p-2 rounded text-white font-bold ${bgColor}`}
      onClick={onClick}
    >
      {title}
    </button>
  );
};

export default Button;
