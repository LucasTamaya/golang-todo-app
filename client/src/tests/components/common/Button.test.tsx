import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Button from "../../../components/common/Button";

const mockedFn = jest.fn();

describe("Button component", () => {
  it("should match the snapshot", () => {
    const component = renderer.create(
      <Button title="Button" bgColor="bg-teal-400" onClick={mockedFn} />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should call the function that we pass in props if we click on the button", () => {
    render(<Button title="Button" bgColor="bg-teal-400" onClick={mockedFn} />);

    fireEvent.click(
      screen.getByRole("button", {
        name: /button/i,
      })
    );

    expect(mockedFn).toHaveBeenCalledTimes(1);
  });
});
