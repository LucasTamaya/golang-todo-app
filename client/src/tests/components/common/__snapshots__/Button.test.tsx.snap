// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match the snapshot 1`] = `
<button
  className="p-2 rounded text-white font-bold bg-teal-400"
  onClick={[MockFunction]}
>
  Button
</button>
`;
